sdk/pinocchio/src/entrypoint/mod.rs

```diff
@@ -200,38 +198,31 @@ pub unsafe fn deserialize<'a, const MAX_ACCOUNTS: usize>(
             }
         }
 
-        // process any remaining accounts to move the offset to the instruction
+        // Process any remaining accounts to move the offset to the instruction
         // data (there is a duplication of logic but we avoid testing whether we
-        // have space for the account or not)
+        // have space for the account or not).
         for _ in processed..total_accounts {
-            let account_info: *mut Account = input.add(offset) as *mut _;
+            let account_info: *mut Account = input as *mut Account;
+            // rent epoch (8 bytes)
+            input = input.add(core::mem::size_of::<u64>());
```

Comment:
Maybe we can let this one slide, but I'd again prefer to have the line in both branches

Reply:
I checked with @LucasSte and in this case it seems better to not have the `else` block since we "save" jumps when an account do not satisfy the `if`-path. I will update the comment to make it more clear.